tuf/data/roles.go

```diff
@@ -101,6 +113,111 @@ type Role struct {
 	Email            string   `json:"email,omitempty"`
 }
 
+// BaseRole is an internal representation of a root/targets/snapshot/timestamp role, with its public keys included
+type BaseRole struct {
```

Comment:
After talking with @cyli and @endophage, I think the plan is to leave these as-is for this PR.  These will be an intermediate step before we introduce individual root/targets/snapshot/timestamp roles with the tree walker